.github/workflows/deploy-pipeline.yml

```
@@ -0,0 +1,411 @@
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v4
+
+    - name: Run Performance Tests
+      run: |
+        # Install k6 for load testing
+        sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys C5AD17C747E3415A3642D57D77C6C491D6AC1D69
+        echo "deb https://dl.k6.io/deb stable main" | sudo tee /etc/apt/sources.list.d/k6.list
+        sudo apt-get update
+        sudo apt-get install k6
+        
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)